File: api/src/main/java/com/api/app/entity/OrderGoods.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("OrderGoods")
@Getter
@Setter
public class OrderGoods extends SystemEntity {
    private static final long serialVersionUID = 4567890123456789012L;

    @Schema(description = "주문번호")
    private String orderNo;

    @Schema(description = "상품번호")
    private String goodsNo;

    @Schema(description = "단품번호")
    private String itemNo;

    @Schema(description = "판매가")
    private Long salePrice;

    @Schema(description = "공급원가")
    private Long supplyPrice;

    @Schema(description = "상품명")
    private String goodsName;

    @Schema(description = "단품명")
    private String itemName;
}

File: api/src/main/kotlin/com/api/app/entity/OrderGoods.kt
package com.api.app.entity

import jakarta.persistence.*
import java.io.Serializable

@Embeddable
data class OrderGoodsId(
    @Column(name = "ORDER_NO", length = 15) val orderNo: String = "",
    @Column(name = "GOODS_NO", length = 15) val goodsNo: String = "",
    @Column(name = "ITEM_NO", length = 3) val itemNo: String = ""
) : Serializable

@Entity
@Table(name = "ORDER_GOODS")
class OrderGoods : SystemEntity() {

    @EmbeddedId
    var id: OrderGoodsId = OrderGoodsId()

    @Column(name = "SALE_PRICE", nullable = false, columnDefinition = "numeric")
    var salePrice: Long = 0

    @Column(name = "SUPPLY_PRICE", nullable = false, columnDefinition = "numeric")
    var supplyPrice: Long = 0

    @Column(name = "GOODS_NAME", nullable = false, length = 200)
    var goodsName: String = ""

    @Column(name = "ITEM_NAME", nullable = false, length = 100)
    var itemName: String = ""

    // Convenience accessors
    val orderNo: String get() = id.orderNo
    val goodsNo: String get() = id.goodsNo
    val itemNo: String get() = id.itemNo
}
